packages/client/lib/net/protocol/ethprotocol.ts

```diff
@@ -121,6 +121,12 @@ export class EthProtocol extends Protocol {
       ],
       decode: ([reqId, bodies]: [Buffer, BlockBodyBuffer[]]) => [new BN(reqId), bodies],
     },
+    {
+      name: 'NewBlock',
+      code: 0x07,
+      encode: ([block, td]: [Block, BN]) => [block.serialize(), td.toBuffer()],
+      decode: ([block, td]: [BlockBuffer, Buffer]) => [Block.fromValuesArray(block), new BN(td)],
```

Comment:
I was actually able to add an integration test for the first time here https://github.com/ethereumjs/ethereumjs-monorepo/pull/1441/commits/5a152ec2a9afb53552cccf9c57d3bb626a7fdd80 for the `Transactions` method respectively I digged into the test setup code base for the first time and roughly followed upon what is done there.

You can add an analogue test for the `NewBlock` method.

```typescript
peer.eth!.send('Transactions', [tx])
t.pass('handled Transactions')
```

This doesn't seem like much but it is actually already a huge win since it at least ensures that this code is running through and doesn't throw with an exception on decoding, which is the most likely thing which can happen here I guess (some room for other additional bugs though as well here, but at least some good start).

Reply:
Okay, I'll do you one better and not just trigger the event but re-write the test to pass once the service received the decoded event :-)